packages/logger/src/Logger.ts

```diff
@@ -493,27 +511,31 @@ class Logger extends Utility implements LoggerInterface {
   }
 
   /**
-   * Alias for removePersistentLogAttributes.
+   * It removes attributes based on provided keys to all log items generated by this Logger instance.
    *
    * @param {string[]} keys
    * @returns {void}
    */
   public removeKeys(keys: string[]): void {
```

Comment:
I think `removeKeys` should remove only from the `temporaryLogAttributes`, just like `appendKeys` adds only to that.

Reply:
I have a concern about this, in the [docs](https://docs.powertools.aws.dev/lambda/typescript/latest/core/logger/#appending-persistent-additional-log-keys-and-values) we remove a persistent attribute with `removeKeys()`
And we have a [test case](https://github.com/aws-powertools/powertools-lambda-typescript/blob/5522e2818e55706ee27b8d6f153b789396b0c156/packages/logger/tests/unit/Logger.test.ts#L1386-L1412). Isn't it a breaking change?

Reply:
True.

We'll have to change the docs and consider the current behavior as a regression/bug that was not intended, and the new behavior the correct one that we meant to support all along.